pandas/tests/groupby/test_groupby.py

```diff
@@ -1736,3 +1736,18 @@ def test_groupby_multiindex_series_keys_len_equal_group_axis():
     expected = pd.Series([3], index=ei)
 
     assert_series_equal(result, expected)
+
+
+def test_groupby_groups_in_BaseGrouper():
+    # GH 26326
+    mi = pd.MultiIndex.from_product([['A', 'B'],
+                                     ['C', 'D']], names=['alpha', 'beta'])
+    df = pd.DataFrame({'foo': [1, 2, 1, 2], 'bar': [1, 2, 3, 4]},
+                      index=mi)
+    grp1 = df.groupby([pd.Grouper(level='alpha'), 'beta'])
```

Comment:
can you call this

result = 

and the nbggrp1  -> expected

then do then consecutively, e.g. like

```
result = df.groupby(....)
expected = ....

assert result.groups == expected.groups

result = df......
expected = ....

assert

Reply:
sure, looks cleaner that way.